src/main/java/pvs/app/controller/GitlabApiController.java

```diff
@@ -0,0 +1,119 @@
+        ObjectMapper objectMapper = new ObjectMapper();
+        List<GitlabCommitDTO> gitlabCommitDTOs = gitlabCommitService.getAllCommits(repoOwner, repoName);
+        String gitlabCommitDTOsJson;
+
+        try {
+            gitlabCommitDTOsJson = objectMapper.writeValueAsString(gitlabCommitDTOs);
+            return ResponseEntity.status(HttpStatus.OK)
+                    .body(gitlabCommitDTOsJson);
+        } catch (JsonProcessingException e) {
+            logger.debug(e.getMessage());
+            e.printStackTrace();
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(exceptionMessage);
+        }
+    }
+
+    @PostMapping("/gitlab/{username}/{password}")
+    public ResponseEntity<String> oauth2login(@PathVariable("username") String username, @PathVariable("password") String password) throws GitLabApiException {
+        System.out.println("going to execute gitlab api..");
+        this.gitlabApiService.oauth2login(username, password);
+        return ResponseEntity.status(HttpStatus.OK).body("log in succeed");
+    }
+
+    @PostMapping("/gitlab/commits/{repoOwner}/{repoName}")
+    public ResponseEntity<String> getCommitsFromGitlab(@PathVariable("repoOwner") String repoOwner, @PathVariable("repoName") String repoName) {
+        System.out.println("going to get commit from gitlab...");
+        try{
+            if(this.gitlabApiService.getCommitsFromGitlab(repoOwner, repoName)){
+                return ResponseEntity.status(HttpStatus.OK).body("get commit from gitlab succeed");
+            }else{
+                return ResponseEntity.status(HttpStatus.OK).body("get commit from gitlab failed");
+            }
+
+        }catch (InterruptedException | GitLabApiException e){
+            logger.debug(e.getMessage());
+            e.printStackTrace();
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(exceptionMessage);
+        }
+
+    }
+
+    @GetMapping("/gitlab/issues/{repoOwner}/{repoName}")
+    public ResponseEntity<String> getIssuesFromGitlab(@PathVariable("repoOwner") String repoOwner, @PathVariable("repoName") String repoName) {
+        System.out.println("going to get issue from gitlab...");
+        ObjectMapper objectMapper = new ObjectMapper();
+        List<GitlabIssueDTO> gitlabIssueDTOs;
+
+        try {
+            gitlabIssueDTOs = gitlabApiService.getIssuesFromGitlab(repoOwner, repoName);
+            if (null == gitlabIssueDTOs) {
+                System.out.println("test1");
+                return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                        .body("cannot get issue data");
+            }
+        } catch (InterruptedException | GitLabApiException e) {
+            System.out.println("test2");
+            logger.debug(e.getMessage());
+            e.printStackTrace();
+            Thread.currentThread().interrupt();
```

Comment:
Could you please explain why there should interrupt the current thread?
Suggest remove this line if it is just a mistake.

Reply:
It uses the getIssuesFromGitlab function which puts data into GitlabIssueLoaderThread and runs by multithread.
When it is running, it may has some error. That's why the interrupt function is here.